### PR TITLE
Added mod to numpy frontend mathematical functions

### LINKED ISSUE #6582: mod
mod - Add mathematical function to numpy frontend https://github.com/unifyai/ivy/issues/1525

### PR DESCRIPTION
**Pull request by salisbuk7897**
Closes #6582

### CODE REVIEW

File: .idea/ivy.iml [hunk 1]
```diff
@@ -1,7 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <module type="PYTHON_MODULE" version="4">
   <component name="NewModuleRootManager">
-    <content url="file://$MODULE_DIR$" />
+    <content url="file://$MODULE_DIR$">
```
Comment (kurshakuz) [inline .idea/ivy.iml:4]: Please don't update files in the .idea folder. Can you remove those update from the PR?

File: .idea/misc.xml [hunk 2]
```diff
@@ -7,4 +7,4 @@
   <component name="PyPackaging">
     <option name="earlyReleasesAsUpgrades" value="true" />
   </component>
-</project>
+</project>
```
Comment (kurshakuz) [inline .idea/misc.xml:10]: Same here

File: .idea/vcs.xml [hunk 3]
```diff
@@ -2,7 +2,5 @@
 <project version="4">
   <component name="VcsDirectoryMappings">
     <mapping directory="$PROJECT_DIR$" vcs="Git" />
-    <mapping directory="$PROJECT_DIR$/ivy_tests/test_array_api" vcs="Git" />
```
Comment (kurshakuz) [inline .idea/vcs.xml:5]: Here as well

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_arithmetic_operations.py [hunk 4]
```diff
@@ -434,3 +434,57 @@ def test_numpy_floor_divide(
         subok=True,
         test_values=False,
     )
+
+
```
Comment (kurshakuz) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_arithmetic_operations.py:438]:
```suggestion

# mod
```

Comment (kurshakuz) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_arithmetic_operations.py:438]: I think we can just add this small comment for consistency

**Review by kurshakuz — CHANGES_REQUESTED** [review]
Hey! Thank you for your PR! I just checked it locally and it is passing the test. Can you please add the changes I requested and then it is good to go.

File: .idea/ivy.iml (hunk 1 above)
Comment (salisbuk7897) [reply .idea/ivy.iml:4]: I didn't update them manually. i remember running  ./merge_with_upstream.sh master. in my git bash to update the master branch. when created a commit, I saw 5 files changed. 
thanks for the detailed correction. I will make the changes now

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_arithmetic_operations.py (hunk 4 above)
Comment (salisbuk7897) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_arithmetic_operations.py:438]: Comment Added

**Comment by salisbuk7897** [comment]
I have reverted the changes and hopefully this time aroud i did everything correctly.

Thanks

**Review by kurshakuz — APPROVED** [review]
Thank you! Now LGTM!

**Pull request merged**